homeassistant/components/sql/sensor.py

```diff
@@ -223,6 +225,11 @@ def __init__(
                 name=name,
             )
 
+    @property
+    def available(self) -> bool:
+        """Return available only if query returns proper values."""
+        return self.state not in [STATE_UNAVAILABLE, STATE_UNKNOWN]
+
```

Comment:
I think this is too specific; the SQL sensor allows deriving a sensor's state from any SQL query, there's no good reason why the sensor should unconditionally be set to unavailable just because the SQL query returned `"unavailable"` or `"unknown"`.

Maybe the validation should happen in `SQLSensor.update` instead by setting `self._attr_native_value` to `None` if `self._numeric_state_expected` returns `True` but the state is not numeric?

Reply:
I think we leave this one until we sorted `command_line` and probably do same here to get a template to be used for availability instead.